### PR TITLE
Created the basic layout for presenting a restaurant list

### PR DESCRIPTION
**Pull request by ktamir**
Created components for a basic restaurant list with mock data. Redux and server integration to come next :)
Screenshot:
https://drive.google.com/open?id=1C0yjkTsQj8ogXQWpe9ePQQscoW1X5-UU

### CODE REVIEW

File: client/src/flowTypes.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// @flow
+
+export type Restaurant = {
+  name: string,
+  cuisine: string,
+  rating: number,
+  address: string,
+  deliveryTime: number,
```
Comment (NoyaArie) [inline client/src/flowTypes.js]: so `deliveryTime` is mandatory?

File: client/src/AppHeader/AppHeader.jsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
```
Comment (NoyaArie) [inline client/src/AppHeader/AppHeader.jsx]: // @flow?

File: client/src/AppHeader/AppHeader.jsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import { Input } from 'antd';
+import { Layout } from 'antd';
```
Comment (NoyaArie) [inline client/src/AppHeader/AppHeader.jsx]: `import { Input, Layout } from 'antd';`

File: client/src/AppHeader/AppHeader.scss [hunk 4]
```diff
@@ -0,0 +1,25 @@
+.app-header {
+  background-image: url('pizza_shoot.png');
+  height: 250px;
+  text-align: center;
+
+  .main-title, .secondary-title {
+    color: #ffffff;
```
Comment (NoyaArie) [inline client/src/AppHeader/AppHeader.scss]: can you extract this to const?

File: client/src/flowTypes.js (hunk 1 above)
Comment (NoyaArie) [reply client/src/flowTypes.js]: please read about the difference between `deliveryTime?: number` and `deliveryTime: ?number`, and read about exact types {| .... |}

File: client/src/flowTypes.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+  accepts10bis: boolean
```
Comment (NoyaArie) [inline client/src/flowTypes.js]: `acceptsTenBis`, `accepts10Bis`

File: client/src/Filter/Filter.jsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+export const Filter = () => {
+  // Array with values ranging from 30 to 120 with differences of 10 (30, 40, 50, ..., 120)
+  const speedRange = [...Array(10)].map((_, i) => 30 + i * 10);
+
+  return (<div className="filter">
+    <div className="filter-col">
+      <div className="filter-type">Cuisine</div>
+      <Select className="filter-select" placeholder="Italian, French, Burgers..." allowClear={true} size="large">
+        <Option value="italian">Italian</Option>
```
Comment (NoyaArie) [inline client/src/Filter/Filter.jsx]: lets extract to array and use map to add options :)

File: client/src/Filter/Filter.jsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+import './Filter.scss';
+import { Select } from 'antd';
+
+const Option = { Select };
+
+export const Filter = () => {
+  // Array with values ranging from 30 to 120 with differences of 10 (30, 40, 50, ..., 120)
+  const speedRange = [...Array(10)].map((_, i) => 30 + i * 10);
+
+  return (<div className="filter">
```
Comment (NoyaArie) [inline client/src/Filter/Filter.jsx]: after ( new line

File: client/src/Filter/Filter.jsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        <Option value="italian">Italian</Option>
+        <Option value="french">French</Option>
+        <Option value="burgers">Burgers</Option>
+      </Select>
+    </div>
+
+    <div className="filter-col">
+      <div className="filter-type">Rating</div>
+      <Select className="filter-select" placeholder="1 to 3" allowClear={true} size="large">
+        <Option value="1">1</Option>
```
Comment (NoyaArie) [inline client/src/Filter/Filter.jsx]: ^^ :)

File: client/src/Filter/Filter.scss [hunk 9]
```diff
@@ -0,0 +1,20 @@
+$filter-background-color: #f54343;
+  .filter {
```
Comment (NoyaArie) [inline client/src/Filter/Filter.scss]: indentation

File: client/src/Filter/Filter.scss [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+    background-color: $filter-background-color;
+    color: #ffffff;
```
Comment (NoyaArie) [inline client/src/Filter/Filter.scss]: use the const  :)

File: client/src/Filter/Filter.scss [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+    display: flex;
+    flex-direction: row;
+    justify-content: center;
+    padding-bottom: 10px;
+    .filter-col {
```
Comment (NoyaArie) [inline client/src/Filter/Filter.scss]: add an empty line before (also in `.filter-type` and `.filter-select`)

File: client/src/Filter/Filter.jsx [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+      </Select>
+    </div>
+
+    <div className="filter-col">
+      <div className="filter-type">Speed</div>
+      <Select className="filter-select" placeholder="30 to 120 minutes" allowClear={true} size="large">
+        {speedRange.map(number => <Option key={number}>Up to {number} minutes</Option>)}
+      </Select>
+    </div>
+  </div>);
```
Comment (NoyaArie) [inline client/src/Filter/Filter.jsx]: new line before )

File: client/src/RestaurantList/RestaurantCard.jsx [hunk 13]
```diff
@@ -0,0 +1,21 @@
+// @flow
+
+import { Card } from 'antd';
+import React from 'react';
+import './RestaurantCard.scss';
+import type { Restaurant } from '../flowTypes';
+
+type Props = {
+  restaurant: Object<Restaurant>,
```
Comment (NoyaArie) [inline client/src/RestaurantList/RestaurantCard.jsx]: why not `restaurant: Restaurant`

File: client/src/RestaurantList/RestaurantCard.jsx [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+export const RestaurantCard = (props: Props) => {
+  const { name, accepts10bis, rating } = props.restaurant;
+  return (<Card className="restaurant-card">
+    <div className="card-details">
+      <h3>{name}</h3>
+      {accepts10bis && <span className="ten-bis">10bis</span>}
+      <div>Rating: {Array.from({ length: rating }, (_, i) => <span key={i}>⭐</span>)}</div>
```
Comment (NoyaArie) [inline client/src/RestaurantList/RestaurantCard.jsx]: if the rating is not an Integer?

File: client/src/RestaurantList/RestaurantList.jsx [hunk 15]
```diff
@@ -0,0 +1,35 @@
+// @flow
+import React, { Component } from 'react';
+import { RestaurantCard } from './RestaurantCard';
+import type { Restaurant } from '../flowTypes';
+
+type State = Array<Restaurant>
+
+class RestaurantList extends Component<{}, State> {
```
Comment (NoyaArie) [inline client/src/RestaurantList/RestaurantList.jsx]: IMO change the component to get restaurants as props

File: client/src/RestaurantList/RestaurantList.jsx [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+  state = {
+    restaurants: [{
```
Comment (NoyaArie) [inline client/src/RestaurantList/RestaurantList.jsx]: read about faker.js :)

File: client/src/Filter/Filter.jsx [hunk 17]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
```
Comment (NoyaArie) [inline client/src/Filter/Filter.jsx]: // @flow

File: client/src/RestaurantList/RestaurantList.jsx (hunk 15 above)
Comment (ktamir) [reply client/src/RestaurantList/RestaurantList.jsx]: Will be handled when I'll integrate state management :)

File: client/src/RestaurantList/RestaurantList.jsx (hunk 16 above)
Comment (ktamir) [reply client/src/RestaurantList/RestaurantList.jsx]: Cool! Doesn't contain restaurant/cuisines though. The ruby one does, I'll use it on the server side

File: client/src/Filter/Filter.jsx [hunk 18]
```diff
@@ -0,0 +1,42 @@
+// @flow
+import React from 'react';
+import './Filter.scss';
+import { Select } from 'antd';
+
+const Option = { Select };
+
+export const Filter = () => {
```
Comment (NoyaArie) [inline client/src/Filter/Filter.jsx]: so the filter, for now, doesn't do anything?

File: client/src/AppHeader/AppHeader.jsx [hunk 19]
```diff
@@ -0,0 +1,17 @@
+// @flow
+
+import React from 'react';
+import { Input, Layout } from 'antd';
+import './AppHeader.scss';
+
+const { Search } = Input;
+
+export const AppHeader = () => (<Layout.Header className="app-header">
```
Comment (NoyaArie) [inline client/src/AppHeader/AppHeader.jsx]: new line after (

File: client/src/AppHeader/AppHeader.jsx [hunk 20]
```diff
@@ -0,0 +1,17 @@
...
+  <h1 className="main-title">NeatEat</h1>
+  <h3 className="secondary-title">Never get hungry again.</h3>
+  <Search
+    placeholder="Find a restaurant..."
+    size="large"
+    className="search-restaurant-input"
+  />
+</Layout.Header>);
```
Comment (NoyaArie) [inline client/src/AppHeader/AppHeader.jsx]: the new line before the )
and let make sure that the linter is configured correctly in the IDE

File: client/src/flowTypes.js [hunk 21]
```diff
@@ -0,0 +1,10 @@
+// @flow
+
+export type Restaurant = {
+  name?: string,
```
Comment (NoyaArie) [inline client/src/flowTypes.js]: name is not mandatory? 
address? cuisine? rating? accepts10Bis?

File: client/src/RestaurantList/RestaurantCard.scss [hunk 22]
```diff
@@ -0,0 +1,17 @@
+.restaurant-card {
+  width: 300px;
+  height: 100px;
+  h3 {
```
Comment (NoyaArie) [inline client/src/RestaurantList/RestaurantCard.scss]: new line before, also before .ten-bis and .card-details

File: client/src/RestaurantList/RestaurantCard.jsx [hunk 23]
```diff
@@ -0,0 +1,22 @@
...
+  restaurant: Restaurant
+}
+
+export const RestaurantCard = (props: Props) => {
+  const { name, accepts10bis, rating } = props.restaurant;
+  return (<Card className="restaurant-card">
+    <div className="card-details">
+      <h3>{name}</h3>
+      {accepts10bis && <span className="ten-bis">10bis</span>}
+      <div>Rating: {Array.from({ length: rating }, (_, i) =>
```
Comment (NoyaArie) [inline client/src/RestaurantList/RestaurantCard.jsx]: the rating can be zero? the rating can be 1.5?

Comment (ktamir) [reply client/src/RestaurantList/RestaurantCard.jsx]: Rating can be either 1, 2, or 3. It's defined in `flowTypes.js`

File: client/src/flowTypes.js (hunk 21 above)
Comment (ktamir) [reply client/src/flowTypes.js]: Got confused with the `?`, now I got it :)

File: client/src/Filter/Filter.jsx (hunk 18 above)
Comment (ktamir) [reply client/src/Filter/Filter.jsx]: Not yet. It will be implemented with redux selectors on next PRs.

**Pull request merged**